internal/langserver/handlers/did_change_watched_files.go

```diff
@@ -0,0 +1,79 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/creachadair/jrpc2"
+	"github.com/hashicorp/terraform-ls/internal/job"
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	"github.com/hashicorp/terraform-ls/internal/protocol"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+	"github.com/hashicorp/terraform-ls/internal/terraform/ast"
+)
+
+func (svc *service) DidChangeWatchedFiles(ctx context.Context, params lsp.DidChangeWatchedFilesParams) error {
+	var ids job.IDs
+
+	for _, change := range params.Changes {
+		uri := string(change.URI)
+		fi, err := os.Stat(uri)
+		if err != nil {
+			jrpc2.ServerFromContext(ctx).Notify(ctx, "window/showMessage", &lsp.ShowMessageParams{
+				Type:    lsp.Warning,
+				Message: fmt.Sprintf("Unable to update: %s, Failed to open directory", uri),
+			})
+			continue
+		}
+
+		if !fi.IsDir() {
+			jrpc2.ServerFromContext(ctx).Notify(ctx, "window/showMessage", &lsp.ShowMessageParams{
+				Type:    lsp.Warning,
+				Message: fmt.Sprintf("No such directory for URI: %s", uri),
+			})
+			continue
+		}
+
+		filename := path.Base(uri)
+		if !ast.IsModuleFilename(filename) && !ast.IsVarsFilename(filename) {
+			jrpc2.ServerFromContext(ctx).Notify(ctx, "window/showMessage", &lsp.ShowMessageParams{
+				Type:    lsp.Warning,
+				Message: fmt.Sprintf("Unable to update file: %s, filetype not supported", uri),
+			})
+			continue
+		}
+
+		// only handle `Changed` event type
+		if change.Type == protocol.Changed {
+			dh := ilsp.HandleFromDocumentURI(change.URI)
+
+			// check existence
+			_, err = svc.modStore.ModuleByPath(dh.Dir.Path())
+			if err != nil {
+				return err
```

Comment:
I believe we should just warn and move ahead with other changes here as well? Similarly for the block below.

Reply:
I think this one we could just ignore/skip without warning as there can be genuinely valid reason for an error which is outside of user's control.

Reply:
Agreed, the server should still process the other changes as part of the same request. But, should we not log it internally somewhere, skipping/ignoring it just means it's completely suppressed, right?